link/src/test/java/com/stripe/android/link/ui/signup/SignUpViewModelTest.kt

```diff
@@ -0,0 +1,450 @@
+                calledCount += 1
+            }
+        }
+
+        createViewModel(linkEventsReporter = linkEventsReporter)
+
+        assertThat(linkEventsReporter.calledCount).isEqualTo(1)
+    }
+
+    @Test
+    fun `When email is valid then lookup is triggered with delay`() = runTest(dispatcher) {
+        val linkAccountManager = object : FakeLinkAccountManager() {
+            override suspend fun lookupConsumer(email: String, startSession: Boolean): Result<LinkAccount?> {
+                delay(100)
+                return super.lookupConsumer(email, startSession)
+            }
+        }
+        val viewModel = createViewModel(prefilledEmail = null, linkAccountManager = linkAccountManager)
+
+        assertThat(viewModel.contentState.signUpState).isEqualTo(SignUpState.InputtingPrimaryField)
+
+        viewModel.emailController.onRawValueChange("valid@email.com")
+        assertThat(viewModel.contentState.signUpState).isEqualTo(SignUpState.InputtingPrimaryField)
+
+        // Advance past lookup debounce delay
+        advanceTimeBy(SignUpViewModel.LOOKUP_DEBOUNCE + 1.milliseconds)
+
+        assertThat(viewModel.emailController.fieldValue.value).isEqualTo("valid@email.com")
+        assertThat(viewModel.contentState.signUpState).isEqualTo(SignUpState.VerifyingEmail)
+    }
+
+    @Test
+    fun `When multiple valid emails entered quickly then lookup is triggered only for last one`() =
+        runTest(dispatcher) {
+            val linkAccountManager = object : FakeLinkAccountManager() {
+                var calledCount = 0
+                val emails = arrayListOf<String>()
+                override suspend fun lookupConsumer(email: String, startSession: Boolean): Result<LinkAccount?> {
+                    calledCount += 1
+                    emails.add(email)
+                    delay(100)
+                    return super.lookupConsumer(email, startSession)
+                }
+            }
+            val viewModel = createViewModel(prefilledEmail = null, linkAccountManager = linkAccountManager)
+
+            viewModel.emailController.onRawValueChange("first@email.com")
+            advanceTimeBy(SignUpViewModel.LOOKUP_DEBOUNCE / 2)
+
+            viewModel.emailController.onRawValueChange("second@email.com")
+            advanceTimeBy(SignUpViewModel.LOOKUP_DEBOUNCE / 2)
+
+            viewModel.emailController.onRawValueChange("third@email.com")
+            assertThat(viewModel.contentState.signUpState).isEqualTo(SignUpState.InputtingPrimaryField)
+
+            // Advance past lookup debounce delay
+            advanceTimeBy(SignUpViewModel.LOOKUP_DEBOUNCE + 1.milliseconds)
+
+            assertThat(viewModel.contentState.signUpState).isEqualTo(SignUpState.VerifyingEmail)
+
+            assertThat(linkAccountManager.calledCount).isEqualTo(1)
+
+            assertThat(linkAccountManager.emails.size).isEqualTo(1)
+            assertThat(linkAccountManager.emails.firstOrNull()).isEqualTo("third@email.com")
```

Comment:
you might be able to use `assertThat(linkAccountManager.emails).containsExactly("third@email.com")` to simplify these asserts